client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import lyrics from './lyrics';
+import {
+	HELLO_DOLLY_NEXT_LYRIC,
+} from './action-types';
+
+const initialState = {
+	lyricIndex: 0
+};
+
+export default function( state = initialState, action ) {
+	switch ( action.type ) {
+		case HELLO_DOLLY_NEXT_LYRIC:
+			return advanceToNextLyric( state );
+		default:
+			return state;
+	}
+}
```

Comment:
Thanks for the tip here. I'm not used to using tree-based `combineReducers()` calls. If it can simplify our redux state handling, then I'm all for it.